docs: re-add paragraph about `ON DELETE CASCADE` (#2513)
cap-js/docs: guides/databases/cdl-to-ddl.md

#### `ON DELETE CASCADE`

Think of the example above: a book can still exist, even if its author is deleted. However, this differs for existential relationships like the one between a book and its pages: a page cannot exist without a corresponding book. In such cases, you want the pages to be automatically deleted when the book (parent) is deleted.

Typically such existential relationships are modeled in CDS using [compositions](../../cds/cdl#compositions).
That is why for managed **backlink** associations (those used in a composition's on-condition via `$self = <comp>.<backlink>`), the foreign key constraints are generated with `ON DELETE CASCADE` by default, instead of `ON DELETE RESTRICT`:

::: code-group
```cds [CDS Source]
entity Books {
  key ID : Integer;
  pages: Composition of many Pages on pages.book = $self;
}
entity Pages {
  key number: Integer;
  key book: Association to Books; // → ON DELETE CASCADE
}
```
:::

::: code-group

> ```sql [=> &nbsp; Generated DDL]

Reviewer: Or is this really needed?
```suggestion
```sql [=>  Generated DDL]
```

> [!tip] Consider using <code>@assert.target</code> instead
> Database constraints are meant to protect against data corruption due to programming errors. Prefer using the [`@assert.target`](../services/constraints#assert-target) for application-level input validation, which is more tuned for typical application scenarios, with error messages taylored for end users.

#### `ON DELETE CASCADE`

> Think of the example above: a book can still exist, even if its author is deleted. However, this may be different fo existential relationships like the one between a book and its pages: a page cannot exist without a corresponding book. In such cases, you would want the pages to be automatically deleted when the book (parent) is deleted.

Reviewer 2: ```suggestion
Think of the example above: a book can still exist, even if its author is deleted. However, this may be different for existential relationships like the one between a book and its pages: a page cannot exist without a corresponding book. In such cases, you would want the pages to be automatically deleted when the book (parent) is deleted.
```

Typically such existential relationships are modeled in CDS using [compositions](../../cds/cdl#compositions).

> That is why for managed **backlink** associations — those used in a composition's on-condition via `$self = <comp>.<backlink>` 

Reviewer 2: ```suggestion
That is why for managed **backlink** associations — those used in a composition's on-condition via `$self = <comp>.<backlink>` -
```

the foreign key constraints are generated with `ON DELETE CASCADE` by default, instead of `ON DELETE RESTRICT`: 

::: code-group

> ```cds [CDS Source]
> entity Books {
>   pages: Composition of many Pages on pages.book = $self;
> }

Reviewer 2: ```suggestion
entity Books {
  key ID : Integer;
  pages: Composition of many Pages on pages.book = $self;
}
```
the key field is mentioned in the SQL below, so it should also be in the model
